Source/WebCore/page/SecurityOrigin.cpp

```diff
@@ -575,6 +576,9 @@ bool SecurityOrigin::equal(const SecurityOrigin* other) const
 {
     if (other == this)
         return true;
+
+    if (isUnique() || other->isUnique())
+        return m_uniqueOriginIdentifier == other->m_uniqueOriginIdentifier;
```

Comment:
What about the case of two origins with the same m_uniqueOriginIdentifier but from two different processes.
Shouldn't we use a ProcessQualified of a thread safe identifier?

Reply:
How about a UUID?